app/controllers/EnterCustomsProcedureCodeController.scala

```diff
@@ -37,10 +37,16 @@ class EnterCustomsProcedureCodeController @Inject()(identify: IdentifierAction,
                                                     mcc: MessagesControllerComponents,
                                                     view: EnterCustomsProcedureCodeView,
                                                     formProvider: EnterCustomsProcedureCodeFormProvider
-                                     )
+                                                   )
   extends FrontendController(mcc) with I18nSupport {
 
-  private lazy val backLink: Call = controllers.routes.OneCustomsProcedureCodeController.onLoad()
+  private[controllers] def backLink()(implicit request: DataRequest[_]): Call = {
+    if (request.checkMode) {
+      controllers.routes.CheckYourAnswersController.onLoad()
```

Comment:
I think this might need to be using the browser back functionality on this page because if the oneCpc question is changed from CYA then that pages comes to this one and this one will need to return to that one.

Reply:
Updated